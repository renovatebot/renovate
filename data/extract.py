import sys
import json
import os
import distutils.core
from os.path import dirname, realpath

if sys.version_info[:2] >= (3, 3):
  from importlib.machinery import SourceFileLoader
  def load_source(name, path):
    if not os.path.exists(path):
      return {}
    return vars(SourceFileLoader('mod', path).load_module())
else:
  import imp
  def load_source(name, path):
    if not os.path.exists(path):
      return {}
    return vars(imp.load_source('mod', path))

try:
  import setuptools
except ImportError:
  class setuptools:
    def setup():
      pass

try:
  from unittest import mock
except ImportError:
  # for python3.3+
  import mock

@mock.patch.object(setuptools, 'setup')
@mock.patch.object(distutils.core, 'setup')
def invoke(mock1, mock2):
  # Inserting the parent directory of the target setup.py in Python import path:
  sys.path.append(dirname(realpath(sys.argv[-1])))
  # This is setup.py which calls setuptools.setup
  load_source('_target_setup_', sys.argv[-1])
  # called arguments are in `mock_setup.call_args`
  call_args = mock1.call_args or mock2.call_args
  args, kwargs = call_args
  with open('renovate-pip_setup-report.json', 'w', encoding='utf-8') as f:
    json.dump(kwargs, f, ensure_ascii=False, indent=2)

invoke()
