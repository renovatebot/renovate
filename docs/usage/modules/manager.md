# Managers

Renovate is based around the concept of "package managers", or "managers" for short.
These range from traditional package managers like npm, Bundler and Composer through to less traditional concepts like CircleCI or Travis config files.

The goal of Renovate is to detect and maintain all third party dependencies in your repositories, through the use of managers.

## Supported Managers

<!-- Autogenerate in https://github.com/renovatebot/renovatebot.github.io -->
<!-- Autogenerate end -->

## Configuring Managers

### File Matching

Most managers have a default `fileMatch` array.
The `fileMatch` array contains regular expression strings that match against the repository file list.

#### Managers with no default fileMatch

Some managers have no default `fileMatch` regular expression, because they have no filename convention that would let Renovate intelligently filter them.
In such a case, the manager will be disabled until you create a `fileMatch` regular expression, e.g. like the following:

```json
{
  "kubernetes": {
    "fileMatch": ["^config/.*\\.yaml$"]
  }
}
```

#### Extending a manager's default fileMatch

If the default `fileMatch` regular expression for a manager does not match against one of your relevant files, you can _extend_ the existing regular expression(s) by configuring a manager's `fileMatch` like in this example:

```json
{
  "dockerfile": {
    "fileMatch": ["does-not-look-like-a-docker-file"]
  }
}
```

#### Ignoring files that match the default fileMatch

Renovate will _extend_ the existing `fileMatch`, meaning you don't need to include the default regular expressions like `Dockerfile` in your own array.
In other words, the regular expression are "additive".
If a manager matches a file that you _don't_ want it to, ignore it using the `ignorePaths` configuration option.
Also, if you ever find that Renovate is _not_ matching a file name that you're certain it should, check your preset config isn't the cause of it.
The `config:base` preset ignores common test and example directory names, for example.

### Enabling and Disabling Managers

#### Enabling experimental managers

Most managers are enabled by default.
For those that aren't, typically because they are considered experimental, you can enable them opt-in.
For example, if there was a manager called `some-new-manager` then you could enable it this way:

```json
{
  "some-new-manager": {
    "enabled": true
  }
}
```

#### Disabling managers

To disable a specific manager like `gradle`, do this:

```json
{
  "gradle": {
    "enabled": false
  }
}
```

To disable all managers within a language like `python`, do this:

```json
{
  "python": {
    "enabled": false
  }
}
```

Only languages declared by a Renovate manager are supported.
Please check the [list of supported managers](https://docs.renovatebot.com/modules/manager/#supported-managers).

#### Limiting enabled managers

Say you only want to use Renovate for JavaScript packages, and to update your Dockerfile, and don't want any other updates.
You can use the `enabledManagers` array, to list the managers you want to use (`npm`, `dockerfile`):

```json
{
  "enabledManagers": ["npm", "dockerfile"]
}
```

Using the `enabledManager` array disables all other managers, this includes Bundler, Composer, Docker Compose, etc.
