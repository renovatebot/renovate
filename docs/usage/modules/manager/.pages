title: Managers
nav:
  - index.md
# TODO: add all managers ?
#  - Introduction: index.md
#  - ...
collapse_single_pages: true
order: asc
sort_type: natural
