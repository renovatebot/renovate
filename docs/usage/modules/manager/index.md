---
title: Managers
---

# Managers

Renovate is based around the concept of "package managers", or "managers" for short.
These range from traditional package managers like npm, Bundler and Composer through to less traditional concepts like CircleCI or Travis config files.

The goal of Renovate is to detect and maintain all third-party dependencies in your repositories, through the use of managers.

## Supported Managers

<!-- Autogenerate in https://github.com/renovatebot/renovate -->
<!-- Autogenerate end -->

## Configuring Managers

### File Matching

Most Renovate managers have a default `managerFilePatterns` array.
The `managerFilePatterns` array can hold a regular expression or glob pattern, that match against the repository file list.

#### Managers with no default managerFilePatterns

Some managers have no default `managerFilePatterns`, because they have no filename convention that would let Renovate intelligently filter them.
If there is no default `filePattern`, the manager is disabled.
For the manager to work, you must create a `managerFilePatterns` regular expression, or glob pattern.
For example:

```json
{
  "kubernetes": {
    "managerFilePatterns": ["/^config/.*\\.yaml$/"]
  }
}
```

#### Extending a manager's default managerFilePatterns

If the default `managerFilePatterns` for a manager does not match your file(s), you can _extend_ the pattern.
You extend the pattern by configuring the manager's `managerFilePatterns`.
For example:

```json
{
  "dockerfile": {
    "managerFilePatterns": ["does-not-look-like-a-docker-file"]
  }
}
```

#### Ignoring files that match the default managerFilePatterns

Renovate will _extend_ the existing [`managerFilePatterns`](../../configuration-options.md#managerfilepatterns), meaning you don't need to include the default patterns like `Dockerfile` in your own array.
In other words, the patterns are "additive".
If a manager matches a file that you _don't_ want it to, ignore it using the [`ignorePaths`](../../configuration-options.md#ignorepaths) configuration option.
Also, if you ever find that Renovate is _not_ matching a file name that you're certain it should, check your preset config isn't the cause of it.
The `config:recommended` preset ignores common test and example directory names, for example.

### Enabling and disabling managers

#### Enabling experimental managers

Most managers are enabled by default.
For those that aren't, typically because they are considered experimental, you can opt-in manually.
If there was a manager called `some-new-manager` you would enable it like this:

```json
{
  "some-new-manager": {
    "enabled": true
  }
}
```

#### Disabling managers

```json title="Example of disabling a specific manager (gradle)"
{
  "gradle": {
    "enabled": false
  }
}
```

Please check the [list of supported managers](#supported-managers).

#### Limiting enabled managers

Say you only want to use Renovate for JavaScript packages, and to update your Dockerfile, and don't want any other updates.
You can use the `enabledManagers` array, to list the managers you want to use (`npm`, `dockerfile`):

```json
{
  "enabledManagers": ["npm", "dockerfile"]
}
```

Using the `enabledManagers` array disables all other managers, this includes Bundler, Composer, Docker Compose, etc.
