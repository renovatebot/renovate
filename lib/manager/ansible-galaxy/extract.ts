import * as datasourceGalaxy from '../../datasource/galaxy';
import * as datasourceGitTags from '../../datasource/git-tags';
import { logger } from '../../logger';
import { SkipReason } from '../../types';
import type { PackageDependency, PackageFile } from '../types';

const galaxyUrlRegex = /((git\+)?((git|ssh|http(s)?):\/\/)?(.*@)?([\w.-]+)((:\d+)?\/|:))([\w./-]+)(\.git)?/;
const galaxyRoleRegex = /[\w-]+\.[\w-]+/;

function interpretLine(
  lineMatch: RegExpMatchArray,
  lineNumber: number,
  dependency: PackageDependency
): PackageDependency {
  const localDependency: PackageDependency = dependency;
  const key = lineMatch[2];
  const value = lineMatch[3].replace(/["']/g, '');
  switch (key) {
    case 'name': {
      localDependency.managerData.name = value;
      break;
    }
    case 'version': {
      localDependency.managerData.version = value;
      localDependency.currentValue = value;
      localDependency.managerData.lineNumber = lineNumber;
      break;
    }
    case 'scm': {
      localDependency.managerData.scm = value;
      break;
    }
    case 'src': {
      localDependency.managerData.src = value;
      break;
    }
    default: {
      return null;
    }
  }
  return localDependency;
}

function finalize(dependency: PackageDependency): boolean {
  const dep = dependency;
  if (dependency.managerData.version === null) {
    dep.skipReason = SkipReason.NoVersion;
    return false;
  }

  const source: string = dep.managerData.src;
  const sourceMatch: RegExpMatchArray = galaxyUrlRegex.exec(source);
  if (sourceMatch) {
    dep.datasource = datasourceGitTags.id;
    // remove ending `.git` from URLs
    dep.depName = sourceMatch[10].replace(/\.git$/, '');
    // remove leading `git+` from URLs like `git+https://...`
    dep.lookupName = source.replace(/git\+/, '');
  } else if (galaxyRoleRegex.exec(source)) {
    dep.datasource = datasourceGalaxy.id;
    dep.depName = dep.managerData.src;
    dep.lookupName = dep.managerData.src;
  } else if (galaxyRoleRegex.exec(dep.managerData.name)) {
    dep.datasource = datasourceGalaxy.id;
    dep.depName = dep.managerData.name;
    dep.lookupName = dep.managerData.name;
  } else {
    dep.skipReason = SkipReason.NoSourceMatch;
    return false;
  }
  if (dep.managerData.name !== null) {
    dep.depName = dep.managerData.name;
  }

  return true;
}

export default function extractPackageFile(
  content: string
): PackageFile | null {
  logger.trace('ansible-galaxy.extractPackageFile()');
  const newBlockRegEx = new RegExp(/^\s*-\s*((\w+):\s*(.*))$/);
  const blockLineRegEx = new RegExp(/^\s*((\w+):\s*(.*))$/);
  const deps: PackageDependency[] = [];
  const lines = content.split('\n');
  try {
    for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
      let lineMatch = newBlockRegEx.exec(lines[lineNumber]);
      if (lineMatch) {
        const dep: PackageDependency = {
          managerData: {
            name: null,
            version: null,
            scm: null,
            src: null,
          },
        };
        do {
          const localdep = interpretLine(lineMatch, lineNumber, dep);
          if (localdep == null) {
            break;
          }
          const line = lines[lineNumber + 1];

          if (!line) {
            break;
          }
          lineMatch = blockLineRegEx.exec(line);
          if (lineMatch) {
            lineNumber += 1;
          }
        } while (lineMatch);
        if (finalize(dep)) {
          delete dep.managerData;
          deps.push(dep);
        }
      }
    }

    if (!deps.length) {
      return null;
    }
    return { deps };
  } catch (err) /* istanbul ignore next */ {
    logger.debug({ err }, 'Error extracting ansible-galaxy deps');
    return null;
  }
}
