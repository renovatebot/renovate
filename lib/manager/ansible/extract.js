const { getDep } = require('../dockerfile/extract');

module.exports = {
  extractPackageFile,
};

function extractPackageFile(content) {
  logger.debug('ansible.extractPackageFile()');
  let deps = [];
  let lineNumber = 0;
  for (const line of content.split('\n')) {
    const match = line.match(/^\s*image:\s*'?"?([^\s'"]+)'?"?\s*$/);
    if (match) {
      const currentFrom = match[1];
      const dep = getDep(currentFrom);
      logger.debug(
        {
          depName: dep.depName,
          currentValue: dep.currentValue,
          currentDigest: dep.currentDigest,
        },
        'Docker image inside ansible'
      );
      dep.lineNumber = lineNumber;
      dep.versionScheme = 'docker';
      deps.push(dep);
    }
    lineNumber += 1;
  }
  deps = deps.filter(
    dep => !(dep.currentValue && dep.currentValue.includes('${'))
  );
  if (!deps.length) {
    return null;
  }
  return { deps };
}
