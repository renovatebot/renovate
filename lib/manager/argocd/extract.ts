import { loadAll } from 'js-yaml';
import * as gitTags from '../../datasource/git-tags';
import { HelmDatasource } from '../../datasource/helm';
import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
import type { ApplicationDefinition } from './types';
import { fileTestRegex } from './util';

function createDependency(
  definition: ApplicationDefinition
): PackageDependency {
  const source = definition.spec?.source;

  if (source == null) {
    return null;
  }

  // a chart variable is defined this is helm declaration
  if (source.chart) {
    return {
      depName: source.chart,
      registryUrls: [source.repoURL],
      currentValue: source.targetRevision,
      datasource: HelmDatasource.id,
    };
  }
  return {
    depName: source.repoURL,
    currentValue: source.targetRevision,
    datasource: gitTags.id,
  };
}

export function extractPackageFile(
  content: string,
  fileName: string,
  config?: ExtractConfig
): PackageFile | null {
  // check for argo reference. API version for the kind attribute is used
  if (fileTestRegex.test(content) === false) {
    return null;
  }

  const definitions: ApplicationDefinition[] = loadAll(content);

  const deps = definitions
    .map((definition) => createDependency(definition))
    .filter(Boolean);

  return deps.length ? { deps } : null;
}
