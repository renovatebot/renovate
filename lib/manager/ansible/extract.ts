import { logger } from '../../logger';
import * as dockerVersioning from '../../versioning/docker';
import { PackageDependency, PackageFile } from '../common';
import { getDep } from '../dockerfile/extract';

export default function extractPackageFile(
  content: string
): PackageFile | null {
  logger.trace('ansible.extractPackageFile()');
  let deps: PackageDependency[] = [];
  const re = /^\s*image:\s*'?"?([^\s'"]+)'?"?\s*$/;
  for (const line of content.split('\n')) {
    const match = re.exec(line);
    if (match) {
      const currentFrom = match[1];
      const dep = getDep(currentFrom);
      logger.debug(
        {
          depName: dep.depName,
          currentValue: dep.currentValue,
          currentDigest: dep.currentDigest,
        },
        'Docker image inside ansible'
      );
      dep.versioning = dockerVersioning.id;
      deps.push(dep);
    }
  }
  deps = deps.filter((dep) => !dep.currentValue?.includes('${'));
  if (!deps.length) {
    return null;
  }
  return { deps };
}
