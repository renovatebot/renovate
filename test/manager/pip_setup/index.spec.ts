import { readFileSync } from 'fs';
import { exec as _exec } from 'child_process';
import * as extract from '../../../lib/manager/pip_setup/extract';
import { extractPackageFile } from '../../../lib/manager/pip_setup';

const packageFile = 'test/manager/pip_setup/_fixtures/setup.py';
const content = readFileSync(packageFile, 'utf8');

const packageFileJson = 'test/manager/pip_setup/_fixtures/setup.py.json';
const jsonContent = readFileSync(packageFileJson, 'utf8');

const config = {
  localDir: '/tmp/github/some/repo',
};

let processEnv;

const cleanCmd = (str: string) =>
  str
    .split(process.cwd())
    .join('/root/project')
    .replace(/\\(\w)/g, '/$1');

const exec: jest.Mock<typeof _exec> = _exec as any;
jest.mock('child_process');

describe('lib/manager/pip_setup/index', () => {
  describe('extractPackageFile()', () => {
    beforeEach(() => {
      jest.resetAllMocks();
      jest.resetModules();
      extract.resetModule();

      exec.mockImplementationOnce((_cmd, _options, callback) => {
        callback(null, { stdout: '', stderr: 'Python 2.7.17\\n' });
        return undefined;
      });
      exec.mockImplementationOnce((_cmd, _options, callback) => {
        callback(null, { stdout: 'Python 3.7.5\\n', stderr: '' });
        return undefined;
      });
      exec.mockImplementationOnce((_cmd, _options, _callback) => {
        throw new Error();
      });

      processEnv = process.env;
      process.env = {
        HTTP_PROXY: 'http://example.com',
        HTTPS_PROXY: 'https://example.com',
        NO_PROXY: 'localhost',
        HOME: '/home/user',
        PATH: '/tmp/path',
      };
    });
    afterEach(() => {
      process.env = processEnv;
    });
    it('returns found deps', async () => {
      const execCommands = [];
      const execOptions = [];
      exec.mockImplementationOnce((cmd, options, callback) => {
        execCommands.push(cleanCmd(cmd));
        execOptions.push(options);
        callback(null, { stdout: jsonContent, stderr: '' });
        return undefined;
      });
      expect(
        await extractPackageFile(content, packageFile, config)
      ).toMatchSnapshot();
      expect(exec).toHaveBeenCalledTimes(4);
      expect(execCommands).toMatchSnapshot();
      expect(execOptions).toMatchSnapshot();
    });
    it('returns found deps (docker)', async () => {
      jest.resetAllMocks();
      jest.resetModules();
      extract.resetModule();

      const execCommands = [];
      const execOptions = [];

      // docker pull
      exec.mockImplementationOnce((cmd, options, callback) => {
        execCommands.push(cleanCmd(cmd));
        execOptions.push(options);
        callback(null, { stdout: '', stderr: '' });
        return undefined;
      });

      exec.mockImplementationOnce((cmd, options, callback) => {
        execCommands.push(cleanCmd(cmd));
        execOptions.push(options);
        callback(null, { stdout: jsonContent, stderr: '' });
        return undefined;
      });

      expect(
        await extractPackageFile(content, packageFile, {
          ...config,
          binarySource: 'docker',
        })
      ).toMatchSnapshot();
      expect(exec).toHaveBeenCalledTimes(2);
      expect(execCommands).toMatchSnapshot();
      expect(execOptions).toMatchSnapshot();
    });
    it('should return null for invalid file', async () => {
      const execCommands = [];
      const execOptions = [];
      exec.mockImplementationOnce((cmd, options, _callback) => {
        execCommands.push(cleanCmd(cmd));
        execOptions.push(options);
        throw new Error();
      });

      expect(
        await extractPackageFile(
          'raise Exception()',
          '/tmp/folders/foobar.py',
          config
        )
      ).toBeNull();
      expect(exec).toHaveBeenCalledTimes(4);
      expect(execCommands).toMatchSnapshot();
      expect(execOptions).toMatchSnapshot();
    });
    it('catches error', async () => {
      const execCommands = [];
      const execOptions = [];
      jest.resetAllMocks();
      jest.resetModules();
      extract.resetModule();
      exec.mockImplementation((cmd, options, _callback) => {
        execCommands.push(cleanCmd(cmd));
        execOptions.push(options);
        throw new Error();
      });
      expect(
        await extractPackageFile(
          'raise Exception()',
          '/tmp/folders/foobar.py',
          config
        )
      ).toBeNull();
      expect(exec).toHaveBeenCalledTimes(4);
      expect(execCommands).toMatchSnapshot();
      expect(execOptions).toMatchSnapshot();
    });
  });
  /*
  describe('extractSetupFile()', () => {
    it('should return parsed setup() call', async () => {
      expect(
        await extractSetupFile(content, packageFile, config)
      ).toMatchSnapshot();
    });
    it('should support setuptools', async () => {
      expect(
        await extractSetupFile(
          'from setuptools import setup\nsetup(name="talisker")\n',
          await tmpFile(),
          config
        )
      ).toEqual({ name: 'talisker' });
    });
    it('should support distutils.core', async () => {
      expect(
        await extractSetupFile(
          'from distutils.core import setup\nsetup(name="talisker")\n',
          await tmpFile(),
          config
        )
      ).toEqual({ name: 'talisker' });
    });
  });
  */
});
