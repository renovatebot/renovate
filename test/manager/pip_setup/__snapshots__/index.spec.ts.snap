// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/pip_setup/index extractPackageFile() catches error 1`] = `
Array [
  "python --version",
  "python3 --version",
  "python3.8 --version",
  "python \\"/root/project/lib/manager/pip_setup/extract.py\\" \\"/tmp/folders/foobar.py\\"",
]
`;

exports[`lib/manager/pip_setup/index extractPackageFile() catches error 2`] = `
Array [
  Object {
    "encoding": "utf-8",
  },
  Object {
    "encoding": "utf-8",
  },
  Object {
    "encoding": "utf-8",
  },
  Object {
    "cwd": "/tmp/github/some/repo",
    "encoding": "utf-8",
    "timeout": 5000,
  },
]
`;

exports[`lib/manager/pip_setup/index extractPackageFile() returns found deps (docker) 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": ">=3.1.13.0,<5.0",
      "datasource": "pypi",
      "depName": "celery",
      "managerData": Object {
        "lineNumber": 36,
      },
    },
    Object {
      "currentValue": ">=1.7",
      "datasource": "pypi",
      "depName": "logging_tree",
      "managerData": Object {
        "lineNumber": 39,
      },
    },
    Object {
      "currentValue": ">=2.2",
      "datasource": "pypi",
      "depName": "pygments",
      "managerData": Object {
        "lineNumber": 40,
      },
    },
    Object {
      "currentValue": ">=5.0",
      "datasource": "pypi",
      "depName": "psutil",
      "managerData": Object {
        "lineNumber": 41,
      },
    },
    Object {
      "currentValue": ">=3.0",
      "datasource": "pypi",
      "depName": "objgraph",
      "managerData": Object {
        "lineNumber": 42,
      },
    },
    Object {
      "currentValue": ">=1.11.23,<2.0",
      "datasource": "pypi",
      "depName": "django",
      "managerData": Object {
        "lineNumber": 45,
      },
    },
    Object {
      "currentValue": ">=0.11,<2.0",
      "datasource": "pypi",
      "depName": "flask",
      "managerData": Object {
        "lineNumber": 48,
      },
    },
    Object {
      "currentValue": ">=1.4,<2.0",
      "datasource": "pypi",
      "depName": "blinker",
      "managerData": Object {
        "lineNumber": 49,
      },
    },
    Object {
      "currentValue": ">=19.7.0,<20.0",
      "datasource": "pypi",
      "depName": "gunicorn",
      "managerData": Object {
        "lineNumber": 61,
      },
    },
    Object {
      "currentValue": ">=3.2.1,<4.0",
      "datasource": "pypi",
      "depName": "statsd",
      "managerData": Object {
        "lineNumber": 62,
      },
    },
    Object {
      "currentValue": ">=0.15.3,<0.16",
      "datasource": "pypi",
      "depName": "Werkzeug",
      "managerData": Object {
        "lineNumber": 62,
      },
    },
    Object {
      "currentValue": ">=2.10.0,<3.0",
      "datasource": "pypi",
      "depName": "requests",
      "managerData": Object {
        "lineNumber": 63,
      },
      "skipReason": "ignored",
    },
    Object {
      "currentValue": ">=5.27.1,<7.0",
      "datasource": "pypi",
      "depName": "raven",
      "managerData": Object {
        "lineNumber": 64,
      },
    },
    Object {
      "currentValue": ">=0.15.2,<0.17",
      "datasource": "pypi",
      "depName": "future",
      "managerData": Object {
        "lineNumber": 65,
      },
    },
    Object {
      "currentValue": ">=1.0.16,<2.0",
      "datasource": "pypi",
      "depName": "ipaddress",
      "managerData": Object {
        "lineNumber": 66,
      },
    },
  ],
}
`;

exports[`lib/manager/pip_setup/index extractPackageFile() returns found deps (docker) 2`] = `
Array [
  "docker pull renovate/pip",
  "docker run -i --rm -v /tmp/github/some/repo:/tmp/github/some/repo -v /root/project/lib/manager/pip_setup:/root/project/lib/manager/pip_setup -w /tmp/github/some/repo renovate/pip python \\"/root/project/lib/manager/pip_setup/extract.py\\" \\"test/manager/pip_setup/_fixtures/setup.py\\"",
]
`;

exports[`lib/manager/pip_setup/index extractPackageFile() returns found deps (docker) 3`] = `
Array [
  Object {
    "encoding": "utf-8",
  },
  Object {
    "cwd": "/tmp/github/some/repo",
    "encoding": "utf-8",
    "timeout": 5000,
  },
]
`;

exports[`lib/manager/pip_setup/index extractPackageFile() returns found deps 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": ">=3.1.13.0,<5.0",
      "datasource": "pypi",
      "depName": "celery",
      "managerData": Object {
        "lineNumber": 36,
      },
    },
    Object {
      "currentValue": ">=1.7",
      "datasource": "pypi",
      "depName": "logging_tree",
      "managerData": Object {
        "lineNumber": 39,
      },
    },
    Object {
      "currentValue": ">=2.2",
      "datasource": "pypi",
      "depName": "pygments",
      "managerData": Object {
        "lineNumber": 40,
      },
    },
    Object {
      "currentValue": ">=5.0",
      "datasource": "pypi",
      "depName": "psutil",
      "managerData": Object {
        "lineNumber": 41,
      },
    },
    Object {
      "currentValue": ">=3.0",
      "datasource": "pypi",
      "depName": "objgraph",
      "managerData": Object {
        "lineNumber": 42,
      },
    },
    Object {
      "currentValue": ">=1.11.23,<2.0",
      "datasource": "pypi",
      "depName": "django",
      "managerData": Object {
        "lineNumber": 45,
      },
    },
    Object {
      "currentValue": ">=0.11,<2.0",
      "datasource": "pypi",
      "depName": "flask",
      "managerData": Object {
        "lineNumber": 48,
      },
    },
    Object {
      "currentValue": ">=1.4,<2.0",
      "datasource": "pypi",
      "depName": "blinker",
      "managerData": Object {
        "lineNumber": 49,
      },
    },
    Object {
      "currentValue": ">=19.7.0,<20.0",
      "datasource": "pypi",
      "depName": "gunicorn",
      "managerData": Object {
        "lineNumber": 61,
      },
    },
    Object {
      "currentValue": ">=3.2.1,<4.0",
      "datasource": "pypi",
      "depName": "statsd",
      "managerData": Object {
        "lineNumber": 62,
      },
    },
    Object {
      "currentValue": ">=0.15.3,<0.16",
      "datasource": "pypi",
      "depName": "Werkzeug",
      "managerData": Object {
        "lineNumber": 62,
      },
    },
    Object {
      "currentValue": ">=2.10.0,<3.0",
      "datasource": "pypi",
      "depName": "requests",
      "managerData": Object {
        "lineNumber": 63,
      },
      "skipReason": "ignored",
    },
    Object {
      "currentValue": ">=5.27.1,<7.0",
      "datasource": "pypi",
      "depName": "raven",
      "managerData": Object {
        "lineNumber": 64,
      },
    },
    Object {
      "currentValue": ">=0.15.2,<0.17",
      "datasource": "pypi",
      "depName": "future",
      "managerData": Object {
        "lineNumber": 65,
      },
    },
    Object {
      "currentValue": ">=1.0.16,<2.0",
      "datasource": "pypi",
      "depName": "ipaddress",
      "managerData": Object {
        "lineNumber": 66,
      },
    },
  ],
}
`;

exports[`lib/manager/pip_setup/index extractPackageFile() returns found deps 2`] = `
Array [
  "python3 \\"/root/project/lib/manager/pip_setup/extract.py\\" \\"test/manager/pip_setup/_fixtures/setup.py\\"",
]
`;

exports[`lib/manager/pip_setup/index extractPackageFile() returns found deps 3`] = `
Array [
  Object {
    "cwd": "/tmp/github/some/repo",
    "encoding": "utf-8",
    "timeout": 5000,
  },
]
`;

exports[`lib/manager/pip_setup/index extractPackageFile() should return null for invalid file 1`] = `
Array [
  "python3 \\"/root/project/lib/manager/pip_setup/extract.py\\" \\"/tmp/folders/foobar.py\\"",
]
`;

exports[`lib/manager/pip_setup/index extractPackageFile() should return null for invalid file 2`] = `
Array [
  Object {
    "cwd": "/tmp/github/some/repo",
    "encoding": "utf-8",
    "timeout": 5000,
  },
]
`;
